components/Book.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import {
+  Image,
+  StyleSheet,
+  Text,
+  View,
+  TouchableOpacity
+} from 'react-native';
+import { Button} from './common';
+import firebase from 'firebase';
+import { loginDispatch, loginDispatchFalse } from '../redux/actions/authActions'
+import { connect } from 'react-redux';
+
+
+class Book extends Component {
+    componentWillMount(){
+      firebase.auth().onAuthStateChanged((user) => {
+          console.log((this.props, ' in authfirebase', user))
+          if (user) {
+            this.props.loginDispatch(user.uid)
+          }
+          else this.props.loginDispatchFalse()
+        })
+      }
+  
+
+
+  render() {
+    const { imageLinks, title } = this.props.book;
+    const { loggedIn } = this.props.auth;
+    return (
+     <TouchableOpacity style={styles.container} >
+          {/*<View style={styles.imageContainer}>
+            <Image  source={ {uri: imageLinks.thumbnail}} style={styles.image} />
+          </View>*/}            
+          { loggedIn ?  <Button onPress={()=>this.props.saveBook(title)}> Save Book </Button> : <Text>sign in to save</Text>}
+          <Text style={styles.title} numberOfLines={1}>{title}</Text>
+     </TouchableOpacity>
+     
```

Comment:
i think for consistency have a sth like onSave function which calls saveBook? in other components you do it that way.

Reply:
will do